hax/treemakers/posrec.py

```diff
@@ -0,0 +1,163 @@
+import hax
+import numpy as np
+from hax.minitrees import TreeMaker
+from pax.PatternFitter import PatternFitter
+from pax.configuration import load_configuration
+from pax.InterpolatingMap import InterpolatingMap
+from pax import utils
+from pax import exceptions
+from scipy.stats import binom_test
+import json
+from keras.models import model_from_json, Sequential
+from keras.layers import Dense, Dropout, Activation, Reshape, Flatten
+from keras import regularizers
+
+class PositionReconstruction(TreeMaker):
+    """Stores position-reconstruction-related variables.
+    
+    Provides:
+       - s1_pattern_fit: s1 pattern fit computed with corrected
+                         position and areas
+       - s1_pattern_fit_hits: s1 pattern fit computed with corrected
+                         position and hits
+
+    """
+    __version__ = '0.12'
+    extra_branches = ['peaks.area_per_channel[260]',
+                      'peaks.hits_per_channel[260]',
+                      'peaks.n_saturated_per_channel',
+                      'interactions.x','interactions.y','interactions.z']
+
+    def __init__(self):
+        
+        hax.minitrees.TreeMaker.__init__(self)
+        
+        # We need to pull some stuff from the pax config
+        self.pax_config = load_configuration("XENON1T")
```

Comment:
do we also need SR-dependent gains from configs in processing repo?

Reply:
If I understand your question correctly then no. Gains already applied at this stage. This is just to pull some defaults. Unfortunately we wouldn't get override values in case the defaults are overridden in the run doc. I checked that they are not overridden for the values I took but I'm not sure of a better way to generalize.